# CVE-2021-39371


## Metadata
| Published | Last Modified | Assigner | Version | Format |
| --------- | ------------- | -------- | ------- | ------ |
| 2021-08-23 01:15:00 | 2022-06-02 14:48:00 | cve@mitre.org | 4.0 | MITRE | 

## Description
| Language | Description |
| -------- | ----------- |
| en | An XML external entity (XXE) injection in PyWPS before 4.4.5 allows an attacker to view files on the application server filesystem by assigning a path to the entity. OWSLib 0.24.1 may also be affected. |

## Configurations
| Vulnerable | CPE 23 URI | CPE Name | Operator |
| ---------- | ---------- | -------- | -------- |
| True | cpe:2.3:a:osgeo:owslib:0.24.1:*:*:*:*:*:*:* | []  OR |
| True | cpe:2.3:a:osgeo:pywps:*:*:*:*:*:*:*:* | [] | 4.4.5 | OR |
| True | cpe:2.3:o:debian:debian_linux:9.0:*:*:*:*:*:*:* | []  OR |

## CVSSv3 Information
| Score | Severity | Impact Score | Exploitability Score | Vector String | Version | Attack Vector | Attack Complexity | Privileges Required | User Interaction | Scope | Confidentiality Impact | Integrity Impact | Availability Impact |
| ----- | -------- | ------------ | -------------------- | ------------- | ------- | ------------- | ----------------- | ------------------- | ---------------- | ----- | ---------------------- | ---------------- | ------------------- |
| 7.5 | HIGH | 3.6 | 3.9 | CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N | 3.1 | NETWORK | LOW | NONE | NONE | UNCHANGED | HIGH | NONE | NONE |

## CVSSv2 Information
| Score | Severity | Impact Score | Exploitability Score | Vector String | Version | Access Vector | Access Complexity | Authentication | User Interaction Required | Obtain All Privilge | Obtain User Privilege | Obtain Other Privilege | Confidentiality Impact | Integrity Impact | Availability Impact |
| ----- | -------- | ------------ | -------------------- | ------------- | ------- | ------------- | ----------------- | -------------- | ------------------------- | ------------------- | --------------------- | ---------------------- | ---------------------- | ---------------- | ------------------- |
| 5.0 | MEDIUM | 2.9 | 10.0 | AV:N/AC:L/Au:N/C:P/I:N/A:N | 2.0 | NETWORK | LOW | NONE | 0 | False | False | False | PARTIAL | NONE | NONE |

## Problem Type
CWE-611

## References
| Name | Reference URL | Reference Source | Tags |
| ---- | ------------- | ---------------- | ---- |
https://github.com/geopython/pywps/pull/616|https://github.com/geopython/pywps/pull/616|MISC|['Patch', 'Third Party Advisory']
| Name | Reference URL | Reference Source | Tags |
| ---- | ------------- | ---------------- | ---- |
https://github.com/geopython/OWSLib/issues/790|https://github.com/geopython/OWSLib/issues/790|MISC|['Issue Tracking', 'Patch', 'Third Party Advisory']
| Name | Reference URL | Reference Source | Tags |
| ---- | ------------- | ---------------- | ---- |
[debian-lts-announce] 20210904 [SECURITY] [DLA 2754-1] pywps security update|https://lists.debian.org/debian-lts-announce/2021/09/msg00001.html|MLIST|['Mailing List', 'Third Party Advisory']
